frontend/src/utils/api.ts

```diff
@@ -295,42 +296,49 @@ const api = {
     add: user => post("/users/", user),
     update: user => patch(`/users/${user.id}/`, user),
     updateSelf: user => patch(`/users/update_self/`, user),
-    list: (options, abort) => get("/users/", options, { abort }),
+    list: (options, abort?) => get("/users/", options, { abort }),
     listRevisions: (userId, options = {}) => get(`/revisions/`, { user_id: userId, ...options }),
     listVersions: (userId, options = {}) => get(`/versions/`, { revision__user: userId, ...options }),
   },
 
   workflows: {
     get: workflowId => get(`/workflows/${workflowId}/`),
-    list: (options, abort) => get('/workflows/', options, { abort })
+    list: (options, abort?) => get('/workflows/', options, { abort })
   },
 
   groups: {
-    list: (options, abort) => get("/groups/", options, { abort }),
+    list: (options, abort?) => get("/groups/", options, { abort }),
   },
 
   query: {
     search: q => get("/query/search/", { q }, { abort: true }),
   },
 
   sample_lineage: {
-    get: sampleId => get(`/sample-lineage/${sampleId}/graph/`)
+    get: (sampleId: FMSId) => get<JsonResponse>(`/sample-lineage/${sampleId}/graph/`)
   }
 };
 
 
 export default api;
 
-export function withToken(token, fn) {
-  return (...args) => fn(...args)(undefined, () => ({ auth: { tokens: { access: token } } }))
+type WithTokenFn<R extends ResponseWithData<any>> = (...args: any[]) => (_: undefined, getState: () => AuthTokensAccess) => Promise<R>
+export function withToken<R extends ResponseWithData<any>>(token: string | undefined, fn: WithTokenFn<R>) {
```

Comment:
won't this logoff someone if you call it without token ? Why would you call withToken withoutToken ??

Reply:
I see that you could already call withToken with an undefined token, but it might actually be a good thing to prevent it and to get some linter error ?